### PR TITLE
added scripts

### PR DESCRIPTION
**Pull request by Abbas4tech**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    alert("You would be dead but the bonus life saved you!");
+  }
+
+  if (monsterHealth <= 0 && playerHealth > 0) {
+    alert("You won!");
+    reset();
+    writeLog(logEventGameOver, "Player Won!", monsterHealth, playerHealth);
+  } else if (playerHealth <= 0 && monsterHealth > 0) {
+    alert("You lost!");
+    reset();
```
Comment (Tauqeer-Ahmed-99) [inline app.js:130]: Parantheses???

File: app.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    writeLog(logEventGameOver, "Monster Won!", monsterHealth, playerHealth);
+  } else if (playerHealth <= 0 && monsterHealth <= 0) {
+    alert("You have a draw!");
+    reset();
```
Comment (Tauqeer-Ahmed-99) [inline app.js:134]: Is this function being called? @Abbas4tech

**Review by Tauqeer-Ahmed-99 — CHANGES_REQUESTED** [review]

File: app.js (hunk 1 above)
Comment (Abbas4tech) [reply app.js:130]: its there @Tauqeer-Ahmed-99

File: app.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+const modeNormalAttack = "normalAttack";
+const modeStrongAttack = "strongAttack";
+const logEventPlayerAttack = "playerAttack";
+const logEventStrongAttack = "strongAttack";
+const logEventMonsterAttack = "monsterAttack";
+const logEventPlayerHeal = "playerHeal";
+const logEventGameOver = "gameOver";
+
+let battleLog = [];
```
Comment (Rehmatulla-Khan) [inline app.js:26]: can we use `const` here @Abbas4tech ?

File: index.html [hunk 4]
```diff
@@ -5,7 +5,7 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <link rel="stylesheet" href="style.css" />
-    <script src="app.js"></script>
+    <script src="app.js" defer></script>
```
Comment (Rehmatulla-Khan) [inline index.html:8]: why defer can you explain @Abbas4tech ?

File: app.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+const attackBtn = document.getElementById("attack-btn");
+const strongAttackBtn = document.getElementById("strong-attack-btn");
+const healBtn = document.getElementById("heal-btn");
+const logBtn = document.getElementById("log-btn");
+
+const enteredHealth = prompt("Choose maximum life for you and monster!", "100");
+
+let chosenMaxLife = parseInt(enteredHealth);
+
+if (isNaN(chosenMaxLife || chosenMaxLife <= 0)) {
```
Comment (Tanvir04khan) [inline app.js]:
```suggestion
if (isNaN(chosenMaxLife) || chosenMaxLife <= 0) {
```
isNaN function must be closed before "||" this sign

File: index.html (hunk 4 above)
Comment (Abbas4tech) [reply index.html:8]: When the browse hits the `script` tag , defer downloads the script file simultaneously parsing the html and only executes it when the parsing of html completed.

File: app.js (hunk 3 above)
Comment (Abbas4tech) [reply app.js:26]: Yes

File: app.js (hunk 5 above)
Comment (Abbas4tech) [reply app.js:14]: Thankyou for your suggestion. i am changing it @Tanvir04khan

Comment (Abbas4tech) [reply app.js]: Thankyou for your suggestion. i made those changes and fixed the issue. can you please see the commits and review it @Tanvir04khan

File: app.js (hunk 2 above)
Comment (Abbas4tech) [reply app.js:134]: Yes!!! @Tauqeer-Ahmed-99

**Review by Tanvir04khan — APPROVED** [review]

**Review by Tauqeer-Ahmed-99 — APPROVED** [review]
LGTM

**Review by mujeeb100 — APPROVED** [review]

**Review by Rehmatulla-Khan — APPROVED** [review]

**Review by mmusk — COMMENTED** [review]
LGTM

**Pull request merged**